tests/functional/Ticket.php

```diff
@@ -560,6 +560,402 @@ public function testUpdateTicketWithActors(array $actors_input, array $expected_
+     * It then adds a second anonymous requester to the ticket and checks the notification queue again.
+     *
+     * @return void
+     */
+    public function testUpdateActorsNotification(): void
+    {
+        /** @var \DBmysql $DB */
+        global $DB, $CFG_GLPI;
+
+        $this->login();
+
+        $user_db_id = getItemByTypeName(User::class, TU_USER, true);
+        $this->integer($user_db_id)->isGreaterThan(0);
+
+        $user_db_email = \UserEmail::getDefaultForUser($user_db_id);
+        $this->string($user_db_email)->isNotEmpty();
+
+        // Create ticket
+        $ticket_id = $this->createItem(
+            \Ticket::class,
+            [
+                'name'        => 'ticket title',
+                'content'     => 'a description',
+                'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+            ]
+        )->fields['id'];
+
+        // Add ticket actors
+        $ticket_actors = [
+            [
+                'users_id' => 0,
+                'tickets_id' => $ticket_id,
+                'type' => CommonITILActor::REQUESTER,
+                'use_notification' => 1,
+                'alternative_email' => 'test@user.com',
+            ],
+            [
+                'users_id'   => 0,
+                'tickets_id' => $ticket_id,
+                'type' => CommonITILActor::OBSERVER,
+                'use_notification' => 1,
+                'alternative_email' => 'test2@user.com'
+            ],
+            [
+                'users_id'   => $user_db_id,
+                'tickets_id' => $ticket_id,
+                'type' => CommonITILActor::ASSIGN,
+                'use_notification' => 1,
+                'default_email' => $user_db_email,
+            ]
+        ];
+
+        $actors = ['requester' => [], 'observer' => [], 'assign' => []];
+
+        foreach ($ticket_actors as $ticket_actor) {
+            $actor = [
+                'itemtype' => 'User',
+                'items_id' => strval($ticket_actor['users_id']),
+                'use_notification' => $ticket_actor['use_notification'],
+            ];
+
+            if (array_key_exists('alternative_email', $ticket_actor)) {
+                $actor['alternative_email'] = $ticket_actor['alternative_email'];
+            }
+
+            switch ($ticket_actor['type']) {
+                case CommonITILActor::REQUESTER:
+                    $actors['requester'][] = $actor;
+                    break;
+                case CommonITILActor::OBSERVER:
+                    $actors['observer'][] = $actor;
+                    break;
+                case CommonITILActor::ASSIGN:
+                    $actors['assign'][] = $actor;
+                    break;
+            }
+        }
+
+        $this->updateItem(
+            \Ticket::class,
+            $ticket_id,
+            [
+                '_actors' => $actors
+            ]
+        );
```

Comment:
You can link the actors during the creation of the ticket, which could simplify the reading. This will impact your `$this->checkActors(...` which follow

```suggestion
        // Create ticket
        $ticket_id = $this->createItem(
            \Ticket::class,
            [
                'name'        => 'ticket title',
                'content'     => 'a description',
                'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
                '_users_id_requester' => 0,
                '_users_id_requester_notif' => [
                    'use_notification' => 1,
                    'alternative_email' => 'test@user.com'
                ],
                '_users_id_observer' => 0,
                '_users_id_observer_notif' => [
                    'use_notification' => 1,
                    'alternative_email' => 'test2@user.com'
                ],
                '_users_id_assign' => $user_db_id,
                '_users_id_assign_notif' => [
                    'use_notification' => 1,
                    'default_email' => $user_db_email
                ]
            ]
        )->fields['id'];
```

Reply:
Applied : https://github.com/glpi-project/glpi/pull/16070/commits/03f6105c12ac4734a1ba5b0afc91d25565dfaf58